app/models/ability.rb

```diff
@@ -0,0 +1,39 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # can :read, Post
+
+    return unless user.present?
+
+    can :manage, :all, author_id: user.id
```

Comment:
- [ ] here you are implying that the user can manage their own posts and comments only, but as the project requires, a user should be able to also read, comment and like post that does not belong to the user, you can make it look like this
```
 def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
```